### PR TITLE
2026 reduced basis use hdf5 by default for saveload and allow basis loading in sequential

### LINKED ISSUE #2026: Reduced basis: use HDF5 by default for save/load and allow basis loading in sequential
- [x]  use HDF5 by default to save/load the reduced basis
- [x] basis loading in sequential

### PR DESCRIPTION
**Pull request by lberti**
- [ ] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/feelpp/feelpp/pulls) for the same change?
- [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
- [ ] Have you written new tests for your changes?
- [X] Have you successfully run the mor testsuite with your changes locally?
- [ ] Have you written Doxygen comments in your contribution ?

-----

### CODE REVIEW

File: mor/apps/crbonlinerun.cpp [hunk 1]
```diff
@@ -61,7 +61,8 @@ loadPlugin( std::string const& name, std::string const& id )
     auto plugin = factoryCRBPlugin( pluginname, pluginlibname, dirname );
     std::cout << "Loaded the plugin " << plugin->name() << std::endl;
     bool loadFiniteElementDatabase = boption(_name="crb.load-elements-database");
-
+    if(boption("export-solution"))
```
Comment (vincentchabannes) [inline mor/apps/crbonlinerun.cpp]: use _name=

File: mor/mor/feel/feelmor/crb.hpp [hunk 2]
```diff
@@ -1514,6 +1514,7 @@ class CRB : public CRBDB,
         {}
 
     virtual void saveRB();
+    virtual void savenewRBelement();
```
Comment (vincentchabannes) [inline mor/mor/feel/feelmor/crb.hpp]: keep saveRB but you can add in arg if necessary

File: mor/mor/feel/feelmor/crb.hpp [hunk 3]
```diff
@@ -2571,6 +2572,15 @@ CRB<TruthModelType>::offline()
                 std::cout << "save Mesh : " << meshFilename << std::endl;
             M_model->rBFunctionSpace()->saveMesh( meshFilename );
         }
+        // save the rb function space
+        if ( true )
+        {
+            std::string functionSpaceFilenameBase = (boost::format("%1%_functionSpace_p%2%.json")%this->name() %this->worldComm().size()).str();
```
Comment (vincentchabannes) [inline mor/mor/feel/feelmor/crb.hpp]: replace boost format by fmt

File: mor/mor/feel/feelmor/crb.hpp [hunk 4]
```diff
@@ -11617,7 +11641,7 @@ CRB<TruthModelType>::saveJson()
                 ptree.add_child( "scmM", ptreeParameterScmM );
             }
         }
-
+        std::cout << "to write json" << std::endl;
```
Comment (vincentchabannes) [inline mor/mor/feel/feelmor/crb.hpp]: remove or use LOG(INFO)

File: mor/mor/feel/feelmor/crbdb.cpp [hunk 5]
```diff
@@ -102,13 +102,15 @@ CRBDB::lookForDB() const
 {
     //std::cout << "db localpath " << this->dbLocalPath() <<" db filename=" << this->dbFilename() << std::endl;
     // look in local repository $HOME/feel/db/crb/...
-    if ( fs::exists( this->dbLocalPath() / this->dbFilename() ) )
```
Comment (vincentchabannes) [inline mor/mor/feel/feelmor/crbdb.cpp]: maybe be to revert

**Comment by lberti** [comment]
@vincentchabannes, the checks have passed. The last changes might have to be reviewed and, if everything is ok, this PR could be merged

**Review by vincentchabannes — APPROVED** [review]

**Pull request merged**